commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls

```diff
@@ -1,349 +1,242 @@
-    request.setHeader('Content-Type', 'application/json');
-    request.setBody(requestBody);
-    HttpResponse response = http.send(request);
-
-    // If the request is successful, parse the JSON response.
-    if (response.getStatusCode() == 200) {
-      Map<String, Object> resultsFromExternalService = (Map<String, Object>) JSON.deserializeUntyped(
-        response.getBody()
-      );
-      return populateTax(resultsFromExternalService);
-    } else {
-      throw new CalloutException(
-        'There was a problem with the request. Error: ' + response.getStatusCode()
-      );
-    }
-  }
-
-  private Map<String, TaxDataFromExternalService> populateTax(Map<String, Object> resultsFromExternalService){
-    Map<String, TaxDataFromExternalService> taxDetailsFromExternalService = new Map<String, TaxDataFromExternalService>();
-    for (String cartItemId : resultsFromExternalService.keySet()) {
-      Map<String, Object> rateAndAmountFromExternalService = (Map<String, Object>) resultsFromExternalService.get(
-        cartItemId
-      );
-      List<Object> cipaList = (List<Object>) rateAndAmountFromExternalService.get(
-        'itemizedPromotionTaxAmounts'
-      );
-      List<CartAdjustment> cipaObj = new List<CartAdjustment>();
-
-      for (Object cipa : cipaList) {
-        cipaObj.add(
-          new CartAdjustment(
-            (String) ((Map<String, Object>) cipa).get('id'),
-            (Decimal) ((Map<String, Object>) cipa).get('taxAmount')
-          )
-        );
-      }
-      taxDetailsFromExternalService.put(
-        cartItemId,
-        new TaxDataFromExternalService(
-          (Decimal) rateAndAmountFromExternalService.get('rate'),
-          (Decimal) rateAndAmountFromExternalService.get('amount'),
-          (String) rateAndAmountFromExternalService.get('taxName'),
-          (Decimal) rateAndAmountFromExternalService.get('adjustmentTaxAmount'),
-          (Decimal) rateAndAmountFromExternalService.get('totalItemizedPromotionTaxAmount'),
-          cipaObj,
-          (Decimal) rateAndAmountFromExternalService.get('grossUnitPrice'),
-          (Decimal) rateAndAmountFromExternalService.get('netUnitPrice')
-        )
-      );
+        return null;
     }
-    return taxDetailsFromExternalService;
-
-  }
-
-  private Map<String, TaxDataFromExternalService> getTaxesFromStaticResponse(Map<String, CartExtension.CartItem> cartItemsMap, String state, String country, CartExtension.TaxLocaleTypeEnum taxType) {
 
+    private Map<String, TaxData> getTaxesFromStaticResponse(Map<String, CartExtension.CartItem> cartItemsMap, String state, String country, CartExtension.TaxLocaleTypeEnum taxType) {
         Double taxRate = 0.15;
         String responseJson = '{';
         for (String key : cartItemsMap.keySet()) {
             CartExtension.CartItem cartItem = cartItemsMap.get(key);
-            ID cartItemId = cartItem.getId();
+            String cartItemId = (cartItem.getId()==null) ? key : cartItem.getId();
```

Comment:
Am I missing something? This is not always a cartItemId - sometimes this is deliveryGroupId?

Reply:
not this line key fetched in line 237 contains deliverygroup id

Reply:
Perhaps it is better to rename it to `cartItemIdOrDeliveryGroupId`

Reply:
Okay replacing `key` with `cartItemIdOrDeliveryGroupId` ;